packages/ra-core/src/reducer/admin/references/oneToMany.ts

```diff
@@ -58,8 +58,9 @@ const oneToManyReducer: Reducer<OneToManyState> = (
 };
 
 export const getIds = (state: ReduxState, relatedTo: string) =>
-    state.admin.references.oneToMany[relatedTo] &&
-    state.admin.references.oneToMany[relatedTo].ids;
+    (state.admin.references.oneToMany[relatedTo] &&
+        state.admin.references.oneToMany[relatedTo].ids) ||
+    null;
```

Comment:
if the problem is `undefined` vs `null`, I'd rather you change the `isDataLoaded` to use `!= null` instead of `!== null` - less risk of breaking things.